packages/vite/src/node/server/index.ts

```diff
@@ -342,10 +341,11 @@ export async function createServer(
       return startServer(server, port, isRestart)
     },
     async close() {
-      process.off('SIGTERM', exitProcess)
-
-      if (!middlewareMode && process.env.CI !== 'true') {
-        process.stdin.off('end', exitProcess)
+      if (!middlewareMode) {
+        process.once('SIGTERM', exitProcess)
+        if (process.env.CI !== 'true') {
+          process.stdin.on('end', exitProcess)
+        }
```

Comment:
Shouldn't we use `off` here? 👀 
```suggestion
      if (!middlewareMode) {
        process.off('SIGTERM', exitProcess)
        if (process.env.CI !== 'true') {
          process.stdin.off('end', exitProcess)
        }
```

Reply:
Oops 🙈

Reply:
I think it is good, from time to time we need to introduce random changes in the PRs to see if everyone is paying attention 😅